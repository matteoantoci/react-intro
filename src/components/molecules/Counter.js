import React from 'react';
import {Button} from "../atoms/Button";
import PropTypes from "prop-types";

export const Counter = ({ count, onClick, ...props }) =>
  (
    <div className="App-Counter mb16" {...props}>
      <div>This is a counter molecule</div>
      <Button onClick={onClick}>
        {`User clicked ${count} times`}
      </Button>
    </div>
  );

Counter.propTypes = {
  count: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};
