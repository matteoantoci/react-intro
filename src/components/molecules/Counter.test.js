import React from "react";
import { Counter } from './Counter'
import renderer from 'react-test-renderer';

describe('<Counter />', () => {
  it('renders correctly', () => {
    const count = 999;
    const fakeCallback = () => {};

    const tree = renderer
      .create(<Counter onClick={fakeCallback} count={count} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
