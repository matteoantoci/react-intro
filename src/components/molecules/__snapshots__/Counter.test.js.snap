// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders correctly 1`] = `
<div
  className="App-Counter mb16"
>
  <div>
    This is a counter molecule
  </div>
  <button
    onClick={[Function]}
  >
    User clicked 999 times
  </button>
</div>
`;
